unix/lofs9p.ml

```diff
@@ -352,7 +352,11 @@ module New(Params : sig val root : string list end) = struct
     | exception Not_found -> bad_fid
     | path ->
       let realpath = Path.realpath path in
-      Lwt_unix.unlink realpath
+      Printf.fprintf stderr "Calling unlink on %s\n%!" realpath;
+      Lwt_unix.LargeFile.stat realpath
+      >>= fun stats ->
+      ( if stats.Lwt_unix.LargeFile.st_kind = Lwt_unix.S_DIR
+        then Lwt_unix.rmdir else Lwt_unix.unlink ) realpath
```

Comment:
Hiding internal/transient ENOTDIR+EISDIR failures sounds like a good idea to me.